homeassistant/components/esphome/sensor.py

```diff
@@ -52,6 +52,8 @@ def _state(self) -> Optional[SensorState]:
     @property
     def icon(self) -> str:
         """Return the icon."""
+        if self._static_info.icon == "":
+            return None
         return self._static_info.icon
```

Comment:
This should preferably also be validated before returned. The valid icons is an API and we shouldn't trust libraries here.